From ac6553ff3a21eb048f623081b7b7e31e43d38d77 Mon Sep 17 00:00:00 2001
From: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Date: Wed, 20 May 2026 14:14:02 +0300
Subject: [PATCH] storage/innobase/sync/cache.cc: bump ARMv8 arch to ARMv8.3

Building for ARMv8.3-A and later, fails with:

Assembler messages:
{standard input}:169: Error: selected processor does not support `retaa'
{standard input}:271: Error: selected processor does not support `retaa'

It happens because the pmem_cvap() funciton manually inserts the `.arch
armv8-2.a` clause, making GAS believe that retaa instruction (inserted
by GCC) is invalid. Bump the manually inserted clause to armv8.3-a, to
prevent GAS from choking on the retaa instructions.

Upstream-Status: Inappropriate [upstream ticket https://github.com/MariaDB/server/pull/3677#issuecomment-4497763227]
Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
---
 storage/innobase/sync/cache.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/storage/innobase/sync/cache.cc b/storage/innobase/sync/cache.cc
index c5e67709e22b..104a45cee9ee 100644
--- a/storage/innobase/sync/cache.cc
+++ b/storage/innobase/sync/cache.cc
@@ -85,7 +85,7 @@ static void pmem_cvap(const void* buf, size_t size)
 #if defined __ARM_ARCH && __ARM_ARCH == 9
     __asm__ __volatile__(".arch armv9.4-a\n dc cvap, %0" :: "r"(u) : "memory");
 #else
-    __asm__ __volatile__(".arch armv8.2-a\n dc cvap, %0" :: "r"(u) : "memory");
+    __asm__ __volatile__(".arch armv8.3-a\n dc cvap, %0" :: "r"(u) : "memory");
 #endif
 
   __asm__ __volatile__("dmb ishst" ::: "memory");
